common/perf-test-core/src/main/java/com/azure/perf/test/core/ApiPerfTestBase.java

```diff
@@ -225,6 +234,103 @@ public Mono<Void> runAllAsync(long endNanoTime) {
             .then();
     }
 
+    public CompletableFuture<Void> runAllAsyncWithCompletableFuture(long endNanoTime) {
+        completedOperations = 0;
+        lastCompletionNanoTime = 0;
+        long startNanoTime = System.nanoTime();
+        Semaphore semaphore = new Semaphore(options.getConcurrentTaskLimit()); // Use configurable limit
+
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
```

Comment:
May want to use a `LinkedList` here as there could be hundreds of thousands of runs that happen during testing, which will result in the `ArrayList` resizing frequently.

Reply:
yeah good idea